cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -40,13 +51,18 @@ int getBufferSize(Expr bufferExpr, Element why) {
     result = why.(Expr).getType().(ArrayType).getSize() and
     not exists(bufferVar.getUnspecifiedType().(ArrayType).getSize())
     or
-    exists(Class parentClass, VariableAccess parentPtr |
+    exists(Class parentClass, VariableAccess parentPtr, int bufferSize |
       // buffer is the parentPtr->bufferVar of a 'variable size struct'
       memberMayBeVarSize(parentClass, bufferVar) and
       why = bufferVar and
       parentPtr = bufferExpr.(VariableAccess).getQualifier() and
       parentPtr.getTarget().getUnspecifiedType().(PointerType).getBaseType() = parentClass and
-      result = getBufferSize(parentPtr, _) + bufferVar.getType().getSize() - parentClass.getSize()
+      (
+        if exists(bufferVar.getType().getSize())
+        then bufferSize = bufferVar.getType().getSize()
+        else bufferSize = 0
```

Comment:
Should this be `-1` to match the current extractor behavior? Setting `bufferSize` to `0` in this case sounds much more reasonable, but reading the PR description it sounds like that's not what we'd get currently?

Reply:
Reading code is difficult 😢  It currently outputs `0`, and after the update it'll not output a value (which internally in the extractor is represented as `-1`). Thanks for questioning me.

I'll update the commit message once the DCA experiment is done.

Reply:
So, `0` is correct.

Reply:
Ah, I see. Thanks for the explanation!